stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -115,7 +114,7 @@ private ClientCalls() {}
       return getUnchecked(futureUnaryCall(call, param));
     } catch (Throwable t) {
       call.cancel();
-      throw Throwables.propagate(t);
+      throw t instanceof RuntimeException ? (RuntimeException) t : new RuntimeException(t);
```

Comment:
@carl-mastrangelo put that `call.cancel()` in finally and have separate catch blocks?